packages/core/src/client.ts

```diff
@@ -302,8 +303,20 @@ export abstract class Client {
     return this
   }
 
+  /**
+   * @deprecated Use {@link event} instead.
+   */
   logEvent(data: Record<string, unknown>): void {
-    this.__eventsLogger.logEvent(data)
+    logDeprecatedMethod(this.logger, 'Honeybadger.logEvent', 'Honeybadger.event')
+    this.event('log', data)
+  }
+
+  event(type: string, data: Record<string, unknown>): void {
```

Comment:
Will this handle both method signatures from [the spec](https://www.notion.so/honeybadger/Client-Library-Spec-aa891332f7874196aa0695b6d38dca66?pvs=4#08adddbe8f7940ad902e1aed13019df1)?

```ruby
# signature:
Honeybadger.event(event_type, payload={})
Honeybadger.event(payload={})
```

Reply:
No, but we could do it. Typescript supports function overloading:
```
  event(data: Record<string, unknown>): void;
  event(type: string, data: Record<string, unknown>): void;
  event(type: string | Record<string, unknown>, data?: Record<string, unknown>): void {
    if (typeof type === 'object') {
      data = type
      type = type['event_type'] as string ?? undefined
    }
    this.__eventsLogger.log({
      event_type: type,
      ts: new Date().toISOString(),
      ...data
    })
  }
```
What would be the `event_type` if the function is called with an object only? Should we look for an `event_type` in the object? What if the object does not have an `event_type?` The [ruby gem](https://github.com/honeybadger-io/honeybadger-ruby/blob/580bc6055d146c818bf74e5f1e3a3cee991c4e00/lib/honeybadger/agent.rb#L371-L384) does not check if `event_type` exists.

Reply:
An event_type field is not required, so if there isn't one specified either as the string first argument or in the object, don't worry about it. :)

Reply:
OK, that makes sense. I'll go ahead and commit the snippet above then.